resources/overlay_plugin_api.ts

```diff
@@ -0,0 +1,221 @@
+// OverlayPlugin API setup
+
+import { EventMap, EventType, IOverlayHandler } from '../types/event';
+
+declare global {
+  interface Window {
+    __OverlayCallback: EventMap[EventType];
+    dispatchOverlayEvent: EventMap[EventType];
+    OverlayPluginApi: {
+      ready: boolean;
+      callHandler: (msg: string, cb?: (value: string) => unknown) => void;
+    };
+  }
+}
+
+type Subscriber<T> = {
+  [key in EventType]?: T[];
+};
+type EventParameter = Parameters<EventMap[EventType]>[0];
+type VoidFunc<T> = (...args: T[]) => void;
+
+let wsUrl: RegExpExecArray | null = null;
+let ws: WebSocket | null = null;
+let queue: (
+  { [s: string]: unknown } |
+  [{ [s: string]: unknown }, ((value: string | null) => unknown) | undefined]
+)[] = [];
+let rseqCounter = 0;
+const responsePromises: Record<number, (value: unknown) => void> = {};
+
+const subscribers: Subscriber<VoidFunc<unknown>> = {};
+
+const sendMessage = (
+    msg: { [s: string]: unknown },
+    cb?: (value: string | null) => unknown,
+): void => {
+  if (ws) {
+    if (queue)
+      queue.push(msg);
+    else
+      ws?.send(JSON.stringify(msg));
```

Comment:
```suggestion
      ws.send(JSON.stringify(msg));
```

In this case, ws is always not-null, due to the `if` above?